dash/include/dash/iterator/internal/ContiguousRange.h

```diff
@@ -88,9 +107,9 @@ struct ContiguousRangeIterator {
 
     auto cur_last  = cur_first;
     size_type num_copy_elem = 0;
-    constexpr const int ndim = pattern_type::ndim();
+    constexpr const dim_t ndim = pattern_type::ndim();
     const auto& pattern = m_pos.pattern();
-    const int fast_dim = (pattern.memory_order() == dash::ROW_MAJOR) ? ndim - 1 : 0;
+    const dim_t fast_dim = (pattern.memory_order() == dash::ROW_MAJOR) ? ndim - 1 : 0;
```

Comment:
This can be made `constexpr`, right? Not sure why it wasn't before...

Reply:
the clang compiler complains with constexpr, but it is not really necessary at this point.
gcc is fine with constexpr